app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCaptureFragment/EventCaptureFormFragment.java

```diff
@@ -178,7 +179,10 @@ public void showFields(List<FieldViewModel> updates, String lastFocusItem) {
 
             if (!isEmpty(lastFocusItem))
                 dataEntryAdapter.setLastFocusItem(lastFocusItem);
+
+            Parcelable rvState = binding.formRecycler.getLayoutManager().onSaveInstanceState();
             dataEntryAdapter.swap(updates);
+            binding.formRecycler.getLayoutManager().onRestoreInstanceState(rvState);
```

Comment:
not sure about this solution, I'm afraid of the impact this can have in some devices perhaps. Let me think about some solutions and I'll came back !